src/robusta/core/playbooks/pod_utils/imagepull_utils.py

```diff
@@ -0,0 +1,203 @@
+
+
+class ImagePullBackoffInvestigator:
+    configs = [
+        # Containerd
+        {
+            "err_template": 'failed to pull and unpack image ".*?": failed to resolve reference ".*?": .*?: not found',
+            "reason": ImagePullBackoffReason.RepoDoesntExist
+            | ImagePullBackoffReason.ImageDoesntExist
+            | ImagePullBackoffReason.TagNotFound,
+        },
+        {
+            "err_template": (
+                'failed to pull and unpack image ".*?": failed to resolve reference ".*?": '
+                "pull access denied, repository does not exist or may require authorization: server message: "
+                "insufficient_scope: authorization failed"
+            ),
+            "reason": ImagePullBackoffReason.NotAuthorized | ImagePullBackoffReason.ImageDoesntExist,
+        },
+        {
+            "err_template": (
+                'failed to pull and unpack image ".*?": failed to resolve reference ".*?": '
+                "failed to authorize: failed to fetch anonymous token: unexpected status: 403 Forbidden"
+            ),
+            "reason": ImagePullBackoffReason.NotAuthorized,
+        },
+        # Docker
+        {
+            "err_template": (
+                "Error response from daemon: pull access denied for .*?, "
+                "repository does not exist or may require 'docker login': denied: requested access to the resource is denied"
+            ),
+            "reason": ImagePullBackoffReason.NotAuthorized | ImagePullBackoffReason.ImageDoesntExist,
+        },
+        {
+            "err_template": "Error response from daemon: manifest for .*? not found: manifest unknown: manifest unknown",
+            "reason": ImagePullBackoffReason.TagNotFound,
+        },
+        {
+            "err_template": (
+                'Error response from daemon: Head ".*?": denied: '
+                'Permission "artifactregistry.repositories.downloadArtifacts" denied on resource ".*?" \\(or it may not exist\\)'
+            ),
+            "reason": ImagePullBackoffReason.NotAuthorized,
+        },
+        {
+            "err_template": 'Error response from daemon: manifest for .*? not found: manifest unknown: Failed to fetch ".*?"',
+            "reason": ImagePullBackoffReason.ImageDoesntExist | ImagePullBackoffReason.TagNotFound,
+        },
+    ]
+
+    def __init__(self, pod_name: str, namespace: str):
+        self.pod_name = pod_name
+        self.namespace = namespace
+
+        self.pod_events: EventList = EventList.listNamespacedEvent(
+            self.namespace, field_selector=f"involvedObject.name={self.pod_name}"
+        ).obj
+
+    def investigate(self, container_status: ContainerStatus) -> Optional[ImagePullOffInvestigation]:
+        for pod_event in self.pod_events.items:
+            error_message = self.get_kubelet_image_pull_error_from_event(pod_event, container_status.image)
+            logging.info(f"for {pod_event} got message: {error_message}")
```

Comment:
some logs here, I assume they were for testing

Reply:
actually this was the same code  moved from
playbooks/robusta_playbooks/image_pull_backoff_enricher.py

Reply:
we might not still need it